<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>664</width>
    <height>303</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Turing machine emulator</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <widget class="QSplitter" name="splitter">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <widget class="QWidget" name="widget_2" native="true">
       <property name="minimumSize">
        <size>
         <width>240</width>
         <height>0</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>300</width>
         <height>16777215</height>
        </size>
       </property>
       <layout class="QGridLayout" name="gridLayout">
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item row="0" column="0">
         <widget class="QLabel" name="label">
          <property name="maximumSize">
           <size>
            <width>300</width>
            <height>26</height>
           </size>
          </property>
          <property name="text">
           <string>Commands:</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="1" column="0" colspan="2">
         <widget class="QListWidget" name="commandsList">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Minimum">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>300</width>
            <height>16777215</height>
           </size>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QPushButton" name="removeCommand">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>150</width>
            <height>0</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>16777215</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="text">
           <string>Remove selected</string>
          </property>
          <property name="icon">
           <iconset theme="list-remove">
            <normaloff/>
           </iconset>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QPushButton" name="clearList">
          <property name="text">
           <string>Clear</string>
          </property>
          <property name="icon">
           <iconset theme="edit-clear">
            <normaloff/>
           </iconset>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="widget" native="true">
       <property name="minimumSize">
        <size>
         <width>400</width>
         <height>0</height>
        </size>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout">
        <property name="spacing">
         <number>0</number>
        </property>
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="QGroupBox" name="addCommands">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>400</width>
            <height>130</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>400</width>
            <height>130</height>
           </size>
          </property>
          <property name="title">
           <string>Add command</string>
          </property>
          <property name="flat">
           <bool>false</bool>
          </property>
          <widget class="QComboBox" name="newCommandFromState">
           <property name="geometry">
            <rect>
             <x>10</x>
             <y>50</y>
             <width>85</width>
             <height>31</height>
            </rect>
           </property>
          </widget>
          <widget class="QLabel" name="label_2">
           <property name="geometry">
            <rect>
             <x>10</x>
             <y>20</y>
             <width>67</width>
             <height>21</height>
            </rect>
           </property>
           <property name="text">
            <string>State:</string>
           </property>
          </widget>
          <widget class="QLabel" name="label_3">
           <property name="geometry">
            <rect>
             <x>100</x>
             <y>20</y>
             <width>67</width>
             <height>21</height>
            </rect>
           </property>
           <property name="text">
            <string>Symbol:</string>
           </property>
          </widget>
          <widget class="QLineEdit" name="newCommandFromSymbol">
           <property name="geometry">
            <rect>
             <x>100</x>
             <y>50</y>
             <width>71</width>
             <height>31</height>
            </rect>
           </property>
           <property name="text">
            <string>λ</string>
           </property>
          </widget>
          <widget class="QLabel" name="label_4">
           <property name="geometry">
            <rect>
             <x>180</x>
             <y>50</y>
             <width>31</width>
             <height>31</height>
            </rect>
           </property>
           <property name="font">
            <font>
             <pointsize>18</pointsize>
             <weight>75</weight>
             <bold>true</bold>
            </font>
           </property>
           <property name="text">
            <string>→</string>
           </property>
          </widget>
          <widget class="QLineEdit" name="newCommandSetSymbol">
           <property name="geometry">
            <rect>
             <x>300</x>
             <y>50</y>
             <width>71</width>
             <height>31</height>
            </rect>
           </property>
           <property name="text">
            <string>λ</string>
           </property>
          </widget>
          <widget class="QComboBox" name="newCommandToState">
           <property name="geometry">
            <rect>
             <x>210</x>
             <y>50</y>
             <width>85</width>
             <height>31</height>
            </rect>
           </property>
          </widget>
          <widget class="QLabel" name="label_5">
           <property name="geometry">
            <rect>
             <x>210</x>
             <y>20</y>
             <width>67</width>
             <height>21</height>
            </rect>
           </property>
           <property name="text">
            <string>State:</string>
           </property>
          </widget>
          <widget class="QLabel" name="label_6">
           <property name="geometry">
            <rect>
             <x>300</x>
             <y>20</y>
             <width>67</width>
             <height>21</height>
            </rect>
           </property>
           <property name="text">
            <string>Symbol:</string>
           </property>
          </widget>
          <widget class="QPushButton" name="pushButton">
           <property name="geometry">
            <rect>
             <x>10</x>
             <y>90</y>
             <width>95</width>
             <height>31</height>
            </rect>
           </property>
           <property name="text">
            <string>Add</string>
           </property>
           <property name="icon">
            <iconset theme="list-add">
             <normaloff/>
            </iconset>
           </property>
          </widget>
          <widget class="QRadioButton" name="newCommandGoLeft">
           <property name="geometry">
            <rect>
             <x>210</x>
             <y>90</y>
             <width>81</width>
             <height>26</height>
            </rect>
           </property>
           <property name="text">
            <string>LEFT</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
          <widget class="QRadioButton" name="newCommandGoRight">
           <property name="geometry">
            <rect>
             <x>300</x>
             <y>90</y>
             <width>71</width>
             <height>26</height>
            </rect>
           </property>
           <property name="text">
            <string>RIGHT</string>
           </property>
          </widget>
          <widget class="QWidget" name="currentCellMoving" native="true">
           <property name="geometry">
            <rect>
             <x>150</x>
             <y>90</y>
             <width>40</width>
             <height>30</height>
            </rect>
           </property>
           <property name="minimumSize">
            <size>
             <width>40</width>
             <height>30</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>40</width>
             <height>30</height>
            </size>
           </property>
           <widget class="QToolButton" name="currentCellMoveLeft">
            <property name="geometry">
             <rect>
              <x>0</x>
              <y>0</y>
              <width>20</width>
              <height>22</height>
             </rect>
            </property>
            <property name="text">
             <string>←</string>
            </property>
           </widget>
           <widget class="QToolButton" name="currentCellMoveRight">
            <property name="geometry">
             <rect>
              <x>20</x>
              <y>0</y>
              <width>20</width>
              <height>22</height>
             </rect>
            </property>
            <property name="text">
             <string>→</string>
            </property>
           </widget>
          </widget>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="emulator">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="title">
           <string>Emulator</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_2">
           <property name="leftMargin">
            <number>0</number>
           </property>
           <property name="topMargin">
            <number>9</number>
           </property>
           <property name="rightMargin">
            <number>0</number>
           </property>
           <property name="bottomMargin">
            <number>0</number>
           </property>
           <item>
            <widget class="QWidget" name="widget_3" native="true">
             <layout class="QHBoxLayout" name="horizontalLayout_2">
              <property name="spacing">
               <number>0</number>
              </property>
              <property name="leftMargin">
               <number>0</number>
              </property>
              <property name="topMargin">
               <number>0</number>
              </property>
              <property name="rightMargin">
               <number>0</number>
              </property>
              <property name="bottomMargin">
               <number>0</number>
              </property>
              <item>
               <widget class="QToolButton" name="tbExecute">
                <property name="toolTip">
                 <string>Execute algorithm</string>
                </property>
                <property name="statusTip">
                 <string/>
                </property>
                <property name="whatsThis">
                 <string/>
                </property>
                <property name="text">
                 <string>Execute</string>
                </property>
                <property name="icon">
                 <iconset theme="media-seek-forward">
                  <normaloff/>
                 </iconset>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QToolButton" name="tbStart">
                <property name="toolTip">
                 <string>Start debugging</string>
                </property>
                <property name="text">
                 <string>Start</string>
                </property>
                <property name="icon">
                 <iconset theme="media-playback-start">
                  <normaloff/>
                 </iconset>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QToolButton" name="tbStop">
                <property name="text">
                 <string>...</string>
                </property>
                <property name="icon">
                 <iconset theme="media-playback-stop">
                  <normaloff/>
                 </iconset>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QToolButton" name="tbNext">
                <property name="text">
                 <string>Next step</string>
                </property>
                <property name="icon">
                 <iconset theme="go-next">
                  <normaloff/>
                 </iconset>
                </property>
               </widget>
              </item>
              <item>
               <spacer name="horizontalSpacer_2">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>40</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
              <item>
               <widget class="QToolButton" name="saveTape">
                <property name="minimumSize">
                 <size>
                  <width>0</width>
                  <height>24</height>
                 </size>
                </property>
                <property name="maximumSize">
                 <size>
                  <width>16777215</width>
                  <height>24</height>
                 </size>
                </property>
                <property name="text">
                 <string>save tape</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QToolButton" name="loadTape">
                <property name="minimumSize">
                 <size>
                  <width>0</width>
                  <height>24</height>
                 </size>
                </property>
                <property name="maximumSize">
                 <size>
                  <width>16777215</width>
                  <height>24</height>
                 </size>
                </property>
                <property name="text">
                 <string>load tape</string>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <widget class="QScrollArea" name="scrollArea">
             <property name="minimumSize">
              <size>
               <width>0</width>
               <height>45</height>
              </size>
             </property>
             <property name="frameShape">
              <enum>QFrame::StyledPanel</enum>
             </property>
             <property name="verticalScrollBarPolicy">
              <enum>Qt::ScrollBarAlwaysOff</enum>
             </property>
             <property name="horizontalScrollBarPolicy">
              <enum>Qt::ScrollBarAsNeeded</enum>
             </property>
             <property name="widgetResizable">
              <bool>true</bool>
             </property>
             <property name="alignment">
              <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
             </property>
             <widget class="QWidget" name="scrollAreaWidgetContents">
              <property name="geometry">
               <rect>
                <x>0</x>
                <y>0</y>
                <width>390</width>
                <height>67</height>
               </rect>
              </property>
              <layout class="QGridLayout" name="gridLayout_2">
               <property name="sizeConstraint">
                <enum>QLayout::SetFixedSize</enum>
               </property>
              </layout>
             </widget>
            </widget>
           </item>
          </layout>
          <zorder>scrollArea</zorder>
          <zorder>widget_3</zorder>
         </widget>
        </item>
       </layout>
       <zorder>emulator</zorder>
       <zorder>addCommands</zorder>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>664</width>
     <height>29</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionLoad"/>
    <addaction name="actionSave_commands"/>
    <addaction name="actionExit"/>
   </widget>
   <addaction name="menuFile"/>
  </widget>
  <action name="actionLoad">
   <property name="text">
    <string>Load commands</string>
   </property>
  </action>
  <action name="actionSave_commands">
   <property name="text">
    <string>Save commands</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
